### PR TITLE
Collapse several Search models

### LINKED ISSUE #12150: Search model renames
There are a few more renames to make. But not just renames: the "V2" models should be collapsed with their v1 (no suffix, that is) counterparts and original `@odata.type` retained for round-tripping. When new ones are created, the "V2" equivalent should always be created because the v1s will cause the service to fail when using the new api-version.

Old name | New name | Notes
--- | --- | ---
EdgeNGramTokenFilterV2 | EdgeNGramTokenFilter | Collapse with existing EdgeNGramTokenFilter
KeywordTokenizerV2 | KeywordTokenizer | Collapse with existing KeywordTokenizer (will have separate properties for v1 and "V2")
LuceneStandardTokenizerV2 | LuceneStandardTokenizer | Collapse with existing LucenseStandardTokenizer
NGramTokenFilterV2 | NGramTokenFilter | Collapse with existing NGramTokenFilter
PathHierarchyTokenizerV2 | PathHierarchyTokenizer | (*There is no v1*)
Similarity | SimilarityAlgorithm | Do not rename derivative types
Suggester | SearchSuggester

### PR DESCRIPTION
**Pull request by heaths**
Fixes #12150

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs [hunk 1]
```diff
@@ -12,79 +12,5 @@ namespace Azure.Search.Documents.Indexes.Models
 {
     public partial class EdgeNGramTokenFilter : IUtf8JsonSerializable
     {
-        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs:15]: Did this change at all?  Could we leave the generated version?

File: sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs [hunk 2]
```diff
@@ -12,79 +12,5 @@ namespace Azure.Search.Documents.Indexes.Models
...
-                writer.WriteStringValue(Side.Value.ToSerialString());
-            }
-            writer.WritePropertyName("@odata.type");
-            writer.WriteStringValue(ODataType);
-            writer.WritePropertyName("name");
-            writer.WriteStringValue(Name);
-            writer.WriteEndObject();
-        }
-
-        internal static EdgeNGramTokenFilter DeserializeEdgeNGramTokenFilter(JsonElement element)
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs:40]: This also looks the same from a quick glance?

**Review by tg-msft — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs (hunk 1 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs:15]: Unfortunately, no. Merging types is not supported by the generator, and if I don't completely take over de/serialization all sorts of bad things happen that make the source invalid and the project unable to compile.

File: sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs (hunk 2 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Generated/Models/EdgeNGramTokenFilter.Serialization.cs:40]: It is, by design.

**Pull request merged**